nodejs/eks/cluster.ts

```diff
@@ -140,6 +184,26 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
       }, { parent: parent })).role;
     }
 
+    // Create the EKS cluster IAM Role Provider
+    let creationRole: pulumi.Output<aws.iam.Role>;
+    if (args.creationRole) {
+        creationRole = pulumi.output(args.creationRole);
+    } else {
+        creationRole = createEntityRole(name, parent);
+    }
```

Comment:
Do we want to always create a custom role for creating the cluster?  Or do we want to keep defaulting to using the outer provider?

As written, I suspect this will cause all existing clusters to be replaced - which is probably not an option.

Reply:
> As written, I suspect this will cause all existing clusters to be replaced - which is probably not an option.

I can see how that would cause issues for others, would adding a bool flag to toggle the custom role suffice? How would the outputs respond to that? (i.e. `this.registerOutputs` using undefined vars)

I think this should eventually be default, until AWS allows the creation of EKS clusters with the `system:masters` entity as a param instead of inferring it.

Reply:
I lean towards making this role & provider optional for those who require it, and to continue allowing backwards compatibility.